Code/Framework/AzToolsFramework/Tests/Prefab/PrefabInstanceDomGeneratorTestFixture.cpp

```diff
@@ -0,0 +1,269 @@
+        ASSERT_TRUE(patchPathReference.has_value());
+        AZStd::string patchPathStringBefore = patchPathReference->get().GetString();
+        AZStd::string patchPathString = patchPathStringBefore;
+        patchPathString.insert(0, patchPrefix);
+        patchPathReference->get().SetString(patchPathString.c_str(), patch.GetAllocator());
+
+        // Apply the patch
+        TemplateId targetTemplateId = targetInstance.GetTemplateId();
+        PrefabDom& targetTemplateDomReference = m_prefabSystemComponent->FindTemplateDom(targetTemplateId);
+        AZ::JsonSerializationResult::ResultCode result =
+            PrefabDomUtils::ApplyPatches(targetTemplateDomReference, targetTemplateDomReference.GetAllocator(), patch);
+        ASSERT_TRUE(result.GetOutcome() == AZ::JsonSerializationResult::Outcomes::Success);
+
+        // Revert back to previous path
+        patchPathReference->get().SetString(patchPathStringBefore.c_str(), patch.GetAllocator());
+    }
+
+    void PrefabInstanceDomGeneratorTestFixture::GenerateAndValidateInstanceDom(
+        const Instance& instance, EntityAlias entityAlias, float expectedValue)
+    {
+        // Generate a prefab DOM for the provided instance
+        PrefabDom generatedInstanceDom;
+        m_instanceDomGeneratorInterface->GenerateInstanceDom(generatedInstanceDom, instance);
+
+        // Create an instance from the generated prefab DOM for validation
+        Instance instanceFromDom;
+        ASSERT_TRUE(PrefabDomUtils::LoadInstanceFromPrefabDom(
+            instanceFromDom, generatedInstanceDom, PrefabDomUtils::LoadFlags::UseSelectiveDeserialization));
+
+        // Verify that the worldX value of the provided child entity is coming from the correct template
+        EntityOptionalReference childEntity = FindEntityInNestedInstances(instanceFromDom, entityAlias);
+        ASSERT_TRUE(childEntity.has_value());
+        childEntity->get().Init();
+        childEntity->get().Activate(); // to connect to buses such as transform bus
+        float curXValue = 0.0f;
+        AZ::TransformBus::EventResult(curXValue, childEntity->get().GetId(), &AZ::TransformInterface::GetWorldX);
+        EXPECT_EQ(curXValue, expectedValue);
+
+        // Verify that the parent of the container entity is a valid entity
+        EntityOptionalReference containerEntity = instanceFromDom.GetContainerEntity();
+        ASSERT_TRUE(containerEntity.has_value());
+        containerEntity->get().Init();
+        containerEntity->get().Activate(); // to connect to buses such as transform bus
+        AZ::EntityId parentEntityId;
+        AZ::TransformBus::EventResult(parentEntityId, containerEntity->get().GetId(), &AZ::TransformInterface::GetParentId);
+        EXPECT_TRUE(parentEntityId.IsValid());
+    }
+
+    EntityOptionalReference PrefabInstanceDomGeneratorTestFixture::FindEntityInNestedInstances(
+        Instance& instance,
+        EntityAlias entityAlias)
```

Comment:
Would be good to add some comment. By looking at the code, I think the function name can be `FindEntityInNestedInstancesInHierarchy` since it is recursively searching (BFS)?

But can we reuse these functions in Instance class?

```cpp
//! Gets the entity ids for all entities in hierarchy in the Instance DOM.
//! A callback can be provided to manipulates the entities.
//! Note: These are recursive operations that trace all nested instances.
void GetAllEntityIdsInHierarchy(const AZStd::function<bool(AZ::EntityId)>& callback) const;
void GetAllEntitiesInHierarchy(const AZStd::function<bool(AZStd::unique_ptr<AZ::Entity>&)>& callback);
void GetAllEntitiesInHierarchyConst(const AZStd::function<bool(const AZ::Entity&)>& callback) const;
```

Reply:
I couldn't use the existing API because those methods only handle EntityIds. I want to find an entity by alias but to get an entity's alias you need to know the owning instance.

Reply:
I think in general it might be beneficial to add some API methods that work with entity aliases because EntityIds are not persistent.

Reply:
I see. Looks like the existing functions in _Instance_ won't allow us to get the entity alias because in the callback function we cannot access the nested instances.